### ISSUE TITLE
Show user:// in file system dock

### ISSUE THREAD

**Issue by lentsius-bark** [godotengine/godot-proposals#2362]
### Describe the project you are working on
Rhythm based game controlled by dancepads allowing players to learn new vocabulary through physical activity. Providing the much needed connection of education, physical activity and a game. (like beat saber but with an educational touch) 

### Describe the problem or limitation you are having in your project
I often visit the user directory to inspect and tweak:
- saved profiles (resources)
- vocabulary (editable language and categorized words -- super useful for teachers)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Due to the usefulness of custom resources and common practice of saving data user data, having access to the user:// folder where they reside in the file system dock would speed up my workflow greatly.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
![image](https://user-images.githubusercontent.com/38346332/109300283-9ba85700-7836-11eb-9323-089907401c43.png)

### If this enhancement will not be used often, can it be worked around with a few lines of script?
I haven't touched enough of addon code to know whether this is doable with a few lines of script. If it is, let me know!

### Is there a reason why this should be core and not an add-on in the asset library?
Do addons allow creating something of this sort? Lack of my own knowledge prevents me from answering this core question.

**Comment by Calinou** [comment]
Good idea, but one issue with displaying `user://` is that non-resource files won't show up just like in `res://`. We need to add a toggle to display non-resource files in the FileSystem dock first (and probably always enable this for `user://`).

**Comment by pycbouh** [comment]
> and probably always enable this for user://

Would be a bit confusing if this ends up looking like the mockup in the OP, with two equal folders, but the toggle only affects one folder, though.

**Comment by lentsius-bark** [comment]
@Calinou Agreed!
@pycbouh Given that this feature wouldn't be used by first time users, ability to see the user folder could be toggled somewhere, project settings perhaps?

**Comment by pycbouh** [comment]
I wasn't talking about displaying the folder, I was talking about a filter that would show all files not just valid resources, the one that Calinou mentioned.

**Comment by willnationsdev** [comment]
Perhaps show an icon next to the top-level folder identifying whether it is display only resources or all files, and then you have to toggle it individually with it defaulting to different values for each folder?

**Comment by pycbouh** [comment]
Like a visibility icon in the Scene dock? I like it.

**Comment by DavidCambre** [comment]
So I think this is the place to go after [#25665](https://github.com/godotengine/godot/issues/25665) and [#2025](https://github.com/godotengine/godot-proposals/issues/2025) ?

So to recap the conversation.
One toggle in project settings to see User://
One toggle(system) in FileSystem to unhide (all) files?
- Would this be one toggle for every folder?
- Would this show .gitignore ?
- Would this show .gdignore folders ?
- Would this show project.godot ?

I would like to see LICENCE and README of plugins.

**Comment by hawkerm** [comment]
Yeah, was confused by the File Access doc and this behavior of the editor, as they show a 'dat' file:

```cs
public string LoadFromFile()
{
    using var file = FileAccess.Open("user://save_game.dat", FileAccess.ModeFlags.Read);
    string content = file.GetAsText();
    return content;
}
```
https://docs.godotengine.org/en/stable/classes/class_fileaccess.html

I figured I may want to have a "lvl" level type file or something, but then it disappeared from my FileSystem.

**Comment by AThousandShips** [comment]
My concern with this is that it risks confusing users into thinking this will be files that goes into the exported project, rather than being convenient access to the path

**Comment by arkology** [comment]
> My concern with this is that it risks confusing users into thinking this will be files that goes into the exported project, rather than being convenient access to the path

This could be easily hidden by default and activated by project or editor advanced setting.

**Comment by AThousandShips** [comment]
Having general features disabled by default is generally not a good idea, if it's useful it should be enabled by default

**Comment by GlitchCode24** [comment]
We can make a different dock or menu, showing all files (not only resources) and a warning that these files will not be into the exported project. The files will be writable and if you drag and drop something to scene, it will ask for copying the file to res://

**Comment by dinkumdingo** [comment]
Wouldn't it make more sense to have a separate `UserData` dock? I don't see any benefits from having the `res://` and `user://` folders in the same container.
